dot/parachain/availability-store/messages.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package availability_store
+
+import (
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// AvailabilityStoreMessage represents the possible availability store subsystem message
+type AvailabilityStoreMessage scale.VaryingDataType
+
+// QueryAvailableData query a AvailableData from the AV store
+type QueryAvailableData struct {
+	CandidateHash common.Hash
+	AvailableData AvailableData
+}
+
+// Index returns the index of varying data type
+func (QueryAvailableData) Index() uint {
+	return 0
+}
+
+type QueryDataAvailability struct {
+	CandidateHash common.Hash
+	// TDDO: add oneshot sender
```

Comment:
we might use a channel here to get a response right here, based on whether we need to use response right then and there. But we can safely remove this comment. Same goes for other instances of `// TDDO: add oneshot sender`

Reply:
Stubbed in channels for these senders and removed TODOs.